### ISSUE TITLE
Deleted product attributes still appear in Variations tab, causing variation data loss on save

### ISSUE THREAD

**Issue by igorradovanov** [woothemes/woocommerce#61636]
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

When creating a variable product, if a product attribute is added and then removed (either immediately or at any later step), the deleted attribute remains visible in the Variations tab. This allows users to generate variations using the non-existent attribute. When the product is saved and the page is refreshed, all variations created with the deleted attribute are completely emptied, resulting in data loss.

**Root cause:** When an attribute is removed, the `admin-ajax.php` call is not triggered automatically (unlike other attribute operations), so the Variations tab doesn't synchronize with the current state of product attributes.

### Expected behavior

When a product attribute is removed:
1. An admin-ajax call should be triggered to update the product data
2. The Variations tab should immediately reflect the removal and prevent variation generation using the deleted attribute
3. The "Generate variations" button should only show currently existing attributes
4. Users should not be able to create variations with non-existent attributes

### Actual behavior

When a product attribute is removed:
1. No admin-ajax call is triggered
2. The Variations tab still displays the deleted attribute as available
3. Users can generate variations using the deleted attribute
4. After saving the product and refreshing the page, all created variations are completely emptied, resulting in data loss

### Steps to reproduce

1. Go to **Products → Add New**
2. Set **Product Data** to **Variable product**
3. Go to the **Attributes** tab
4. Click **Add attribute** and create a new attribute (e.g., "Color")
5. Add attribute values (e.g., "Red", "Yellow")
6. Click **Save attributes**
7. **Remove the attribute** by clicking the "Remove" button
8. Navigate to the **Variations** tab
9. Observe that the "Generate variations" dropdown still shows the deleted attribute
10. Click **Generate variations** - variations are created despite the attribute being deleted
11. Save the product
12. Refresh the page
13. **Result:** All variations are now empty/lost

### WordPress Environment

<details><summary>System Status Report</summary>

```
### WordPress Environment ###

WordPress address (URL): [Redacted]
Site address (URL): [Redacted]
WC Version: 10.3.3
Legacy REST API Package Version: The Legacy REST API plugin is not installed on this site.
Action Scheduler Version: ✔ 3.9.3
Log Directory Writable: ✔
WP Version: 6.8.3
WP Multisite: –
WP Memory Limit: 256 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_US
External object cache: –

### Server Environment ###

Server Info: nginx/1.26.1
Server Architecture: Darwin 23.6.0 arm64
PHP Version: 8.1.23
PHP Post Max Size: 1,000 MB
PHP Time Limit: 1200
PHP Max Input Vars: 4000
cURL Version: 8.7.1
(SecureTransport) LibreSSL/3.3.6

SUHOSIN Installed: –
MySQL Version: 8.0.35
Max Upload Size: 300 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ✔
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

[REDACTED]

### Post Type Counts ###

attachment: 5
page: 7
post: 2
product: 5
product_variation: 4
revision: 1
wp_navigation: 1
wpcode: 3

### Security ###

Secure connection (HTTPS): ❌
					Your store is not using HTTPS. Learn more about HTTPS and SSL Certificates.
Hide errors from visitors: ✔

### Active Plugins (1) ###

WooCommerce: by Automattic – 10.3.3

### Inactive Plugins (3) ###

Plugin Check (PCP): by WordPress Performance Team and Plugin Review Team – 1.6.0
Product Roles Manager for WooCommerce: by Igor Radovanov – 1.0.0
WP Crontrol: by John Blackbourn – 1.19.3

### Settings ###

Legacy API Enabled: –
Force SSL: –
Currency: RSD (рсд)
Currency Position: right_space
Thousand Separator: .
Decimal Separator: ,
Number of Decimals: 0
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to WooCommerce.com: –
Enforce Approved Product Download Directories: ✔
HPOS feature enabled: ✔
Order datastore: Automattic\WooCommerce\Internal\DataStores\Orders\OrdersTableDataStore
HPOS data sync enabled: –
Enabled Features: analytics
marketplace
order_attribution
site_visibility_badge
remote_logging
blueprint
point_of_sale
custom_order_tables


### Logging ###

Enabled: ✔
Handler: Automattic\WooCommerce\Internal\Admin\Logging\LogHandlerFileV2
Retention period: 30 days
Level threshold: –
Log directory size: 3 KB

### WC Pages ###

Shop base: #7 - /shop/
Cart: #8 - /cart/ -  Contains the woocommerce/cart block
Checkout: #9 - /checkout/ -  Contains the woocommerce/checkout block
My account: #10 - /my-account/ -  Contains the [woocommerce_my_account] shortcode
Terms and conditions: ❌ Page not set

### Theme ###

Name: Twenty Twenty-Five
Version: 1.3
Author URL: https://wordpress.org
Child Theme: ❌ – If you are modifying WooCommerce on a parent theme that you did not build personally we recommend using a child theme. See: How to create a child theme
Theme type: Block theme
WooCommerce Support: ❌ Not declared

### Templates ###

Overrides: /Users/iradovanov/Local Sites/developer/app/public/wp-content/plugins/woocommerce/templates/block-notices/error.php
/Users/iradovanov/Local Sites/developer/app/public/wp-content/plugins/woocommerce/templates/block-notices/notice.php version 9.5.0 is out of date. The core version is 10.2.0
/Users/iradovanov/Local Sites/developer/app/public/wp-content/plugins/woocommerce/templates/block-notices/success.php

Outdated Templates: ❌
					
					
						Learn how to update					 |
					
						
					
					
						Clear system status theme info cache

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customize-store
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
onboarding
onboarding-tasks
pattern-toolkit-full-composability
product-custom-fields
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
printful
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page
launch-your-store

Disabled Features: product-data-views
experimental-blocks
experimental-iapi-mini-cart
experimental-iapi-runtime
coming-soon-newsletter-template
minified-js
product-pre-publish-modal
settings
async-product-editor-category-field
product-editor-template-system
use-wp-horizon
rest-api-v4

Daily Cron: ✔ Next scheduled: 2025-10-26 16:22:02 +00:00
Options: ✔
Notes: 67
Onboarding: skipped

### Action Scheduler ###

Complete: 77
Oldest: 2025-10-16 20:38:55 +0000
Newest: 2025-10-26 11:05:50 +0000

Failed: 1
Oldest: 2024-04-16 17:27:07 +0000
Newest: 2024-04-16 17:27:07 +0000

Pending: 11
Oldest: 2025-10-26 11:53:22 +0000
Newest: 2025-10-27 10:54:18 +0000


### Status report information ###

Generated at: 2025-10-26 11:09:23 +00:00
```
</details>

### Isolating the problem

- [x] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [x] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [x] I can reproduce this bug consistently using the steps above.

**Comment by igorradovanov** [comment]
https://github.com/user-attachments/assets/b7d8dad1-ebd9-425e-ba55-d37137d0c9b7

**Comment by sunyatasattva** [comment]
Hello @igorradovanov thank you for the report. I can reproduce the issue, but the issue only happens if the new attribute is removed WITHOUT pressing the “Save Attributes” button. While I agree that this is an unexpected behavior for the UX perspective and is inconsistent, I'm not sure this is a high impact bug, so I'm not sure we are going to fix this anytime soon. Always happy to accept PRs, however!
